### ISSUE TITLE
Sidenav upgrade

### ISSUE THREAD

**Issue by souperneon** [mapzen/mapzen-docs-generator#238]
Also in: https://mapzen.com/common/styleguide/js-components.html#side-navigation 

![screen shot 2016-12-15 at 5 46 28 pm](https://cloud.githubusercontent.com/assets/6125224/21245066/8e91f4ce-c2ee-11e6-9df5-f81bd3ea2c02.png)

Current behavior:
- Sub-sections are collapsed and are separate pages
- Only one sub-section that is currently active can be open at any given moment

Desired (new) behavior:
- Sub-sections are open on page load 
- Sub-sections are one long page instead of multiple pages
H1 -  Get Started (main sections)
H2 - Tangram Basics (page names) 
H3 - OpenGL (main sections in a page)
H4-6 - smaller sections in a page (collapsed, but expandable)

**Comment by kkowalsky** [comment]
This [index.md](https://github.com/mapzen/mapzen-docs-generator/blob/katie/tangram-update/src-tangram-new/index.md) has the right heading levels. 

If you look at the [config YML for tangram](https://github.com/mapzen/mapzen-docs-generator/blob/katie/tangram-update/config/tangram.yml) you'll see that some of the hierarchy can be built in MKDocs, but the `index.md` doesn't play nice and wants to always be the first part of the TOC (I think)

**Comment by kkowalsky** [comment]
anything you want to test for the new nav should use the src-tangram-new (Still WIP) folder: https://github.com/mapzen/mapzen-docs-generator/tree/katie/tangram-update

**Comment by migurski** [comment]
I would split this from the new Tangram work. It's a separate thing that we want apply throughout, and mixing it up with the other new stuff will make it harder to roll out.

**Comment by souperneon** [comment]
I agree. I was using the Tangram wireframe to illustrate the type hierarchy and explain how the logic should work with the left nav. 
Once we have a version of it built, testing it with Tangram docs will be good :)

**Comment by hanbyul-here** [comment]
Leaving a note for future:
I spent times with docs-generator, and found out it doesn't know about headers outside of current page context. Page structure from config file is saved as `nav`, accessible from `theme`, but headers are not saved as globla object like `nav`, it is saved in `page` scope which is only accessible from current page.
